port/zos390/omrtime.c

```diff
@@ -63,16 +78,22 @@ omrtime_msec_clock(struct OMRPortLibrary *portLibrary)
 uintptr_t
 omrtime_usec_clock(struct OMRPortLibrary *portLibrary)
 {
-	int64_t microsec = MAXPREC() / 8;
+	int64_t microsec = MAXPREC() / OMRPORT_TIME_HIRES_MICROTIME_DIVISOR;
 	return microsec;
 }
 
 uint64_t
 omrtime_current_time_nanos(struct OMRPortLibrary *portLibrary, uintptr_t *success)
 {
 	*success = 1;
-	uint64_t nanos = MAXPREC() * 125;
-	return nanos;
+
+	const int64_t subnanosec = MAXPREC();
+
+	// calculate nanosec = subnanosec * NUMERATOR / DENOMINATOR via integer arithmetics
+	const int64_t nanosec = muldiv64(subnanosec,
```

Comment:
This calculation is much more expensive now than it was I believe.  There are now 7 math operations instead of 1.  If you made the code `uint64_t nanos = MAXPREC() * OMRPORT_TIME_HIRES_NANOTIME_NUMERATOR / OMRPORT_TIME_HIRES_NANOTIME_DENOMINATOR;' some of the math would become a compile time constant and return back to a single math operation